### ISSUE TITLE
iceprod2 - messaging_url default (Trac #857)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#851]
Be smarter about the messaging_url default, so that it works on systems without unix sockets, or systems that don't define localhost correctly.

Preferred order is:
* ipc://$PWD/unix_socket.sock
* tcp://localhost:5555
* tcp://127.0.0.1:5555
* tcp://{system_ip}:5555

Note that the 5555 port is just an example, and probably not what we should pick (something above 10000 is better).

Possibly leave messaging_url empty and fill on startup with a sensible value?

Migrated from https://code.icecube.wisc.edu/ticket/857
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:28", 
    "description": "Be smarter about the messaging_url default, so that it works on systems without unix sockets, or systems that don't define localhost correctly.\n\nPreferred order is:\n* ipc://$PWD/unix_socket.sock\n* tcp://localhost:5555\n* tcp://127.0.0.1:5555\n* tcp://{system_ip}:5555\n\nNote that the 5555 port is just an example, and probably not what we should pick (something above 10000 is better).\n\nPossibly leave messaging_url empty and fill on startup with a sensible value?", 
    "reporter": "david.schultz", 
    "cc": "david.schultz", 
    "resolution": "fixed", 
    "_ts": "1550067088921308", 
    "component": "iceprod", 
    "summary": "iceprod2 - messaging_url default", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2015-01-07T21:28:05", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150504T17:00:50: david.schultz changed milestone from "" to "IceProd 2.0"

**Comment by icecube-trac** [comment]
Trac update at 20150831T20:43:35: 
* david.schultz changed status from "new" to "accepted"
* david.schultz changed owner from "ddelventhal" to "david.schultz"

**Comment by icecube-trac** [comment]
Trac update at 20150831T20:43:59: 
* david.schultz changed status from "accepted" to "closed"
* david.schultz commented:

> In changeset 137076:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="137076"
> dynamically select messaging url. fixes #851
> ```


* david.schultz changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:28: 
* olivas commented:

> Milestone IceProd 2.0 deleted


* olivas changed milestone from "IceProd 2.0" to ""
